src/Runtime/RedispatchHook.php

```diff
@@ -93,11 +94,17 @@ public function redispatch(InputInterface $input)
         // The options the user provided on the commandline will be included
         // in $redispatchArgs.
         $redispatchOptions = [];
+        if (!TerminalUtils::stdinIsTerminal()) {
+            // Force non-interactive mode for the remote command when redirecting stdin
+            $redispatchOptions['no-interaction'] = true;
```

Comment:
I don't think this is the right way to solve this part of the problem. I'll push another commit in a bit.